### PR TITLE
cli: include commit id from build in version

### PR DESCRIPTION
**Pull request by martinvonz**
This changes the version number reported by `jj version` from "0.7.0" to something like
"0.7.0-24a512683bc921699575b6a953624b05c068d544a". The hash is added if running in a jj repo or a git repo.

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Review by Ralith — APPROVED** [review]
Tidy!

File: build.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        println!("cargo:rustc-env=JJ_VERSION={}-{}", version, git_hash);
+    } else {
+        println!("cargo:rustc-env=JJ_VERSION={}", version);
+    }
+
+    Ok(())
+}
+
+fn get_git_hash() -> Option<String> {
+    if let Ok(output) = Command::new("jj")
```
Comment (ilyagr) [inline build.rs:39]: I'm worried this will prevent people from building jj unless they already have jj installed.

Comment (martinvonz) [reply build.rs:39]: The `if let Ok(_)` should handle that case, no?

Comment (ilyagr) [reply build.rs:39]: Never mind, I think you took care of that. :+1:

**Comment by ilyagr** [comment]
I have `test_global_opts` fail because `jj --version` contains color (I think). If I don't hear from you, I'll try to send in a fix eventually.

I have no idea why it happens for me and not in CI or for you.

image

The failure is at

https://github.com/martinvonz/jj/blob/ed0b23d009cff67fcc53288bd60241692c1bafa1/tests/test_global_opts.rs#L59-L62

File: build.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        .args([
+            "--ignore-working-copy",
+            "log",
+            "--no-graph",
+            "-r=@-",
+            "-T=commit_id",
+        ])
+        .output()
+    {
+        if output.status.success() {
```
Comment (ilyagr) [inline build.rs:49]: I also wonder if, in principle, we should check that the working dir is clean. Not sure if that's worth the trouble.

**Pull request merged**